sdk-workflows/src/main/java/io/dapr/workflows/runtime/DaprWorkflowContextImpl.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.workflows.runtime;
+
+import com.microsoft.durabletask.Task;
+import com.microsoft.durabletask.TaskOrchestrationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.NOPLogger;
+import java.time.Duration;
+
+public class DaprWorkflowContextImpl implements WorkflowContext {
+  private final TaskOrchestrationContext innerContext;
+  private final Logger logger;
+
+  /**
+   * Constructor for DaprWorkflowContextImpl.
+   *
+   * @param context TaskOrchestrationContext
+   * @throws IllegalArgumentException if context is null
+   */
+  public DaprWorkflowContextImpl(TaskOrchestrationContext context) throws IllegalArgumentException {
+    this(context, LoggerFactory.getLogger(WorkflowContext.class));
+  }
+
+  /**
+   * Constructor for DaprWorkflowContextImpl.
+   *
+   * @param context TaskOrchestrationContext
+   * @param logger Logger
+   * @throws IllegalArgumentException if context or logger is null
+   */
+  public DaprWorkflowContextImpl(TaskOrchestrationContext context, Logger logger) throws IllegalArgumentException {
+    if (context == null) {
+      throw new IllegalArgumentException("Context cannot be null");
+    } else {
+      this.innerContext = context;
+    }
+
+    if (logger == null) {
+      throw new IllegalArgumentException("Logger cannot be null");
```

Comment:
Should we just create a logger, if not provided - instead of IllegalArgument?

Reply:
That is the idea behind the [`public DaprWorkflowContextImpl(TaskOrchestrationContext context)`](https://github.com/dapr/java-sdk/pull/857/files/9b2881b830eb6d87ca9de0912346f4b31c817576#diff-d11922cd65de99fb47af4c04136f5cdd1a470855f363d5ac79afd1ce8e5a3404R33-R35) constructor.  I feel that, in that context, if you are calling an overloaded constructor without the required arguments an error is appropriate.

I could be convinced otherwise, but that was the intent.